packages/core/contracts/optimistic-assertor/implementation/sovereign-security-manager/SuperbondSovereignSecurityManager.sol

```diff
@@ -0,0 +1,113 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity 0.8.16;
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "./BaseSovereignSecurityManager.sol";
+import "../../interfaces/OptimisticAssertorInterface.sol";
+
+contract SuperbondSovereignSecurityManager is BaseSovereignSecurityManager, Ownable {
+    struct ArbitrationResolution {
+        bool valueSet;
+        bool resolution;
+    }
+
+    struct ClaimBonding {
+        bool superBondReached;
+        IERC20 currency;
+        uint256 currentBondAmount;
+    }
+
+    // Address of linked requesting contract. Before this is set via setAssertingCaller all assertions will be blocked.
+    address public assertingCaller;
+
+    mapping(bytes32 => ArbitrationResolution) public arbitrationResolutions;
+
+    mapping(IERC20 => uint256) public superBonds; //Superbond amounts for each currency.
+
+    mapping(bytes32 => ClaimBonding) public claimBondings; // Track the bondings for each claim.
+
+    event AssertingCallerSet(address indexed assertingCaller);
+    event SuperBondAmountSet(IERC20 indexed currency, uint256 superBondAmount);
+    event SuperBondReached(bytes32 indexed claimId, IERC20 indexed currency);
+
+    // Setting superBondAmount to 0 will block all assertions for that currency.
+    function setSuperBondAmount(IERC20 currency, uint256 superBondAmount) public onlyOwner {
+        superBonds[currency] = superBondAmount;
+        emit SuperBondAmountSet(currency, superBondAmount);
+    }
+
+    // Set the address of the contract that will be allowed to use Optimistic Assertor.
+    // This can only be set once. We do not set this at constructor just to allow for some flexibility in the ordering
+    // of how contracts are deployed.
+    function setAssertingCaller(address _assertingCaller) public onlyOwner {
+        require(_assertingCaller != address(0), "Invalid asserting caller");
+        require(assertingCaller == address(0), "Asserting caller already set");
+        assertingCaller = _assertingCaller;
+        emit AssertingCallerSet(_assertingCaller);
+    }
+
+    function setArbitrationResolution(
+        bytes32 identifier,
+        uint256 time,
+        bytes memory ancillaryData,
+        bool arbitrationResolution
+    ) public onlyOwner {
+        bytes32 requestId = keccak256(abi.encode(identifier, time, ancillaryData));
+        arbitrationResolutions[requestId] = ArbitrationResolution(true, arbitrationResolution);
+    }
+
+    function processAssertionPolicies(bytes32 assertionId) public override returns (AssertionPolicies memory) {
+        OptimisticAssertorInterface optimisticAssertor = OptimisticAssertorInterface(msg.sender);
```

Comment:
One issue I realize now as this can change state we need to add authentication check so that it can be called from Optimistic Assertor.

For now I will be using Optimistic Assertor address set by owner, but we might consider to discover Optimistic Asserter through Finder

Reply:
Addressed in this commit 658eccb0eed992b5e97dcb588e2b09117c23f012